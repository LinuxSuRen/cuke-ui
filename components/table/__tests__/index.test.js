import assert from "power-assert";
import React from "react";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Table from "../index";

const dataSource = [
  {
    name: "黄瓜",
    count: 1,
    id: 1
  },
  {
    name: "西瓜",
    count: 344,
    id: 2
  },
  {
    name: "香蕉",
    count: 199,
    id: 3
  }
];

const columns = [
  {
    title: "ID",
    dataIndex: "id",
    key: "id"
  },
  {
    title: "名字",
    dataIndex: "name",
    key: "name"
  },
  {
    title: "数量",
    dataIndex: "count",
    key: "count"
  },
  {
    title: "操作",
    dataIndex: "setting",
    key: "setting",
    render: (value, data, index) => <Button>删除</Button>
  }
];

import Button from "../../button";
import Empty from "../../empty";
import Pagination from "../../pagination";
import Spin from "../../spin";
import Checkbox from "../../checkbox";

describe("<Table/>", () => {
  it("should render Table", () => {
    const wrapper = render(<Table columns={columns} dataSource={dataSource} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-table class name", () => {
    const wrapper = shallow(
      <Table columns={columns} dataSource={dataSource} />
    );
    assert(wrapper.find(".cuke-table").length === 1);
    assert(wrapper.find(".cuke-table-origin-table").length === 1);
    assert(wrapper.find(".cuke-table-pagination").length === 1);
    assert(wrapper.find(".cuke-table-tbody").length === 1);
  });
  it("should render empty table", () => {
    const wrapper = shallow(<Table columns={columns} dataSource={[]} />);
    assert(wrapper.find(Empty).length === 1);
    assert(wrapper.find(Pagination).length === 0);
    assert(wrapper.find(".cuke-table-tbody").length === 0);
  });

  it("should find custom render", () => {
    const wrapper = render(<Table columns={columns} dataSource={dataSource} />);
    expect(wrapper.text()).toContain("删除");
  });

  it("should can not render Pagination when pagination is false", () => {
    const wrapper = shallow(
      <Table columns={columns} dataSource={dataSource} pagination={false} />
    );
    assert(wrapper.find(Pagination).length === 0);
  });

  it("should render Spin when is loading", () => {
    const wrapper = shallow(
      <Table columns={columns} dataSource={dataSource} loading />
    );
    assert(wrapper.find(Spin).length === 1);
  });

  it("should render border", () => {
    const wrapper = shallow(
      <Table columns={columns} dataSource={dataSource} bordered />
    );
    assert(wrapper.find(".cuke-table-bordered").length === 1);
  });

  it("should render row selection", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <Table
        columns={columns}
        dataSource={dataSource}
        rowSelection={{
          onChange,
          getCheckboxProps(record) {
            console.log("record: ", record);
          }
        }}
      />
    );
    assert(wrapper.find(Checkbox).length === 4);
    wrapper
      .find(Checkbox)
      .at(0)
      .simulate("change", {
        target: {
          checked: true
        }
      });
    expect(onChange).toHaveBeenCalled();
  });
});
