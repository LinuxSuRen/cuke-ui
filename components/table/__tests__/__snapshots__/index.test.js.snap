// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table/> should render Table 1`] = `
<div
  class="cuke-table"
  columns="[object Object],[object Object],[object Object],[object Object]"
>
  <table
    class="cuke-table-origin-table"
  >
    <thead
      class="cuke-table-thead"
    >
      <tr>
        <th>
          ID
        </th>
        <th>
          名字
        </th>
        <th>
          数量
        </th>
        <th>
          操作
        </th>
      </tr>
    </thead>
    <tbody
      class="cuke-table-tbody"
    >
      <tr>
        <td>
          1
        </td>
        <td>
          黄瓜
        </td>
        <td>
          1
        </td>
        <td>
          <button
            class="cuke-button cuke-button-default cuke-button-normal"
            type="button"
          >
            <span>
              删除
            </span>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          西瓜
        </td>
        <td>
          344
        </td>
        <td>
          <button
            class="cuke-button cuke-button-default cuke-button-normal"
            type="button"
          >
            <span>
              删除
            </span>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          香蕉
        </td>
        <td>
          199
        </td>
        <td>
          <button
            class="cuke-button cuke-button-default cuke-button-normal"
            type="button"
          >
            <span>
              删除
            </span>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="cuke-table-pagination"
  >
    <ul
      class="cuke-pagination cuke-pagination-default"
      current="1"
      pageindex="1"
    >
      <li
        class="cuke-pagination-show-total"
      />
      <li
        class="cuke-pagination-item cuke-pagination-prev cuke-pagination-item-disabled"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <polyline
            points="15 18 9 12 15 6"
          />
        </svg>
      </li>
      <li
        class="cuke-pagination-item cuke-pagination-item-selected"
      >
        1
      </li>
      <li
        class="cuke-pagination-item cuke-pagination-next cuke-pagination-item-disabled"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </li>
    </ul>
  </div>
</div>
`;
